File: semgrep-rules/python/xss/testdata/stored_xss.py
from django.utils.safestring import mark_safe
from markupsafe import Markup
import bleach


def render_profile(profile):
    # ruleid: python-xss-django-stored-content-safe
    return mark_safe(profile.bio)


def render_comment(comment):
    # ruleid: python-xss-django-stored-content-safe
    return mark_safe(comment.rendered_html)


def render_clean_comment(comment):
    # ok: python-xss-django-stored-content-safe
    return mark_safe(bleach.clean(comment.rendered_html))


def render_flask_post(post):
    # ruleid: python-xss-flask-stored-content-markup
    return Markup(post.body)


def render_flask_page(page):
    # ruleid: python-xss-flask-stored-content-markup
    return Markup(page.html)


def render_flask_clean(page):
    # ok: python-xss-flask-stored-content-markup
    return Markup(bleach.clean(page.html))

File: semgrep-rules/java/xss/testdata/StoredXss.java
import java.io.IOException;
import javax.servlet.http.HttpServletResponse;
import org.springframework.http.ResponseEntity;

class StoredXssController {
    String renderPost(Post post) {
        // ruleid: java-xss-stored-content-response
        return post.getContent();
    }

    ResponseEntity<String> renderComment(Comment comment) {
        // ruleid: java-xss-stored-content-response
        return ResponseEntity.ok(comment.getBody());
    }

    void renderProfile(Profile profile, HttpServletResponse response) throws IOException {
        // ruleid: java-xss-stored-content-response
        response.getWriter().write(profile.getRenderedHtml());
    }

    String renderStatic() {
        // ok: java-xss-stored-content-response
        return "<p>trusted</p>";
    }
}

class Post {
    String getContent() { return ""; }
}

class Comment {
    String getBody() { return ""; }
}

class Profile {
    String getRenderedHtml() { return ""; }
}
